src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -285,50 +280,60 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
 
   // Returns true if this OM has an owner, false otherwise.
   bool      has_owner() const;
-  int64_t   owner() const;  // Returns null if DEFLATER_MARKER is observed.
+  int64_t   owner() const;  // Returns NO_OWNER if DEFLATER_MARKER is observed.
   int64_t   owner_raw() const;
 
-  static int64_t owner_for(JavaThread* thread);
-  static int64_t owner_for_oop(oop vthread);
+  // These methods return the value we set in _owner when acquiring
+  // the monitor with the given thread/vthread (tid).
+  static int64_t owner_from(JavaThread* thread);
+  static int64_t owner_from(oop vthread);
 
   // Returns true if owner field == DEFLATER_MARKER and false otherwise.
   bool      owner_is_DEFLATER_MARKER() const;
   // Returns true if 'this' is being async deflated and false otherwise.
   bool      is_being_async_deflated();
-  // Clear _owner field; current value must match old_value.
-  void      release_clear_owner(JavaThread* old_value);
+  // Clear _owner field; current value must match thread's tid.
+  void      release_clear_owner(JavaThread* thread);
   // Simply set _owner field to new_value; current value must match old_value.
   void      set_owner_from_raw(int64_t old_value, int64_t new_value);
+  // Same as above but uses tid of current as new value.
```

Comment:
By `tid` here (and elsewhere) you actually mean `thread->threadObj()->thread_id()` - right?

Reply:
It is `thread->vthread()->thread_id()` but it will match `thread->threadObj()->thread_id()` when there is no virtual thread mounted.

Reply:
but we can't change it there to thread_id because then it would be too confusing.  Since it's used for locking, lock_id seems like a good name.